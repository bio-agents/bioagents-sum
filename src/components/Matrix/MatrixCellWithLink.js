import React from 'react'
import { Link } from 'react-router-dom'
import { Image } from 'react-bootstrap'

const MatrixCellWithLink = ({ linkTo, image, text, numberOfServices }) => {
  return (
    <div className='square light-grey-background greyscale'>
      <Link to={linkTo}>
        <div className='square-content'>
          <div className='table'>
            <div className='table-cell text-big'>
              {image
                ? <Image src={image} className={`responsive-img ${!numberOfServices ? 'grey' : ''}`} />
                : text && <span>{text}</span>}
              <div className='text-little'>{numberOfServices}</div>
            </div>
          </div>
        </div>
      </Link>
    </div>
  )
}

export default MatrixCellWithLink
