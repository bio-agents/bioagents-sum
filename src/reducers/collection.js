import * as R from 'ramda'
import * as Actions from '../constants/actionTypes'
import { configCollection } from '../bioagentsSum/common/helperFunctions'

const initialState = {
  activeCollection: '',
  userEnteredCollection: false,
}

export const collection = (state = initialState, action) => {
  const { type, payload } = action

  switch (type) {
    case Actions.SET_COLLECTION:
      return R.evolve({
        activeCollection: R.always(payload.collection !== '' ? payload.collection : configCollection),
        userEnteredCollection: R.always(payload.collection !== '' ? payload.userEnteredCollection : false),
      })(state)

    default:
      return state
  }
}
