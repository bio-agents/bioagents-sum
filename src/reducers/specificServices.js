import * as R from 'ramda'
import * as Actions from '../constants/actionTypes'
import { pickData } from '../bioagentsSum/services/index'

const initialState = {
  count: 0,
  list: [],
  persistExpiresAt: '',
  serviceLoading: false,
  serviceLoaded: false,
  citationsLoading: false,
  citationsLoaded: false,
}

export const specificServicesWithName = (servicesName = '') =>
  (state = initialState, action) => {
    const { type, payload } = action

    if (payload && payload.name !== servicesName) return state

    switch (type) {
      case Actions.SERVICES_FETCH:
        return R.evolve({
          serviceLoading: R.T,
          serviceLoaded: R.F,
          citationsLoading: R.F,
          citationsLoaded: R.F,
        })(state)

      case Actions.SERVICES_FETCH_SUCCESS: {
        const { count, list } = payload.service
        let finalList = pickData(list)

        if (state.list.length !== 0) {
          const oldListByIds = R.groupBy(R.prop('id'), state.list)
          finalList = R.map(agent => R.mergeDeepLeft(agent, oldListByIds[agent.id] ? oldListByIds[agent.id][0] : {}), finalList)
        }

        return R.evolve({
          count: R.always(count),
          list: R.always(finalList),
          serviceLoading: R.F,
          serviceLoaded: R.T,
          citationsLoading: R.T,
          persistExpiresAt: R.always(Date.now() + 600000), // 86400000 ms == 24 hours
        })(state)
      }

      case Actions.SERVICES_FETCH_FAILURE: {
        return R.evolve({
          serviceLoading: R.F,
          serviceLoaded: R.F,
        })(state)
      }

      case Actions.CITATIONS_FETCH_SUCCESS: {
        const { count, list } = payload.service
        const pickedData = pickData(list)

        return R.evolve({
          count: R.always(count),
          list: R.always(pickedData),
          citationsLoading: R.F,
          citationsLoaded: R.T,
        })(state)
      }

      case Actions.CITATIONS_FETCH_FAILURE:
        return R.evolve({
          citationsLoading: R.F,
          citationsLoaded: R.F,
        })(state)

      case Actions.UPTIME_FETCH_SUCCESS: {
        const { count, list } = payload.service
        const pickedData = pickData(list)

        return R.evolve({
          count: R.always(count),
          list: R.always(pickedData),
        })(state)
      }

      case Actions.UPTIME_FETCH_FAILURE: {
        // todo
      }

      default:
        return state
    }
  }
