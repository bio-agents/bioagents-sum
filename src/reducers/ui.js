import * as R from 'ramda'
import * as Actions from '../constants/actionTypes'

const initialState = {
  showReportPage: false,
}

export const ui = (state = initialState, action) => {
  const { type } = action

  switch (type) {
    case Actions.REPORT_PAGE_CHANGE_VISIBILITY:
      return R.evolve({
        showReportPage: R.not,
      })(state)

    default:
      return state
  }
}
