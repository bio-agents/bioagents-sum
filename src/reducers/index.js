import { combineReducers } from 'redux'
import { routerReducer } from 'react-router-redux'
import { reducer as reduxFormReducer } from 'redux-form'
import { specificServicesWithName } from './specificServices'
import * as R from 'ramda'
import { ui } from './ui'
import { collection } from './collection'
import { getServicesNames } from '../bioagentsSum/common/helperFunctions'

const bioagentsSumApp = () => {
  const reducers = R.map(specificServicesWithName, getServicesNames)
  const servicesReducers = R.zipObj(getServicesNames, reducers)

  return combineReducers(R.merge(servicesReducers, {
    ui,
    collection,
    router: routerReducer,
    form: reduxFormReducer,
  }))
}

export default bioagentsSumApp
