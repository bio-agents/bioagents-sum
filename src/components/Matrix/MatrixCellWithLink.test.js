import React from 'react'
import renderer from 'react-test-renderer'
import { ConnectedRouter } from 'react-router-redux'
import { createStore } from 'redux'
import createHistory from 'history/createBrowserHistory'
import reducers from '../../reducers'
import MatrixCellWithLink from './MatrixCellWithLink'
import { Provider } from 'react-redux'

const store = createStore(reducers(), {})
const history = createHistory()

it('Renders MatrixCellWithLink with link and name correctly', () => {
  const component = renderer.create(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <MatrixCellWithLink linkTo='https://google.com/' name='A cell with link' />
      </ConnectedRouter>
    </Provider>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
