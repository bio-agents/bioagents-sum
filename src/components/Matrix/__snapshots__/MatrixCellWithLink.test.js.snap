exports[`test Renders MatrixCellWithLink with link and name correctly 1`] = `
<div
  className="square light-grey-background greyscale">
  <a
    href="https://google.com/"
    onClick={[Function]}>
    <div
      className="square-content">
      <div
        className="table">
        <div
          className="table-cell text-big">
          <div
            className="text-little" />
        </div>
      </div>
    </div>
  </a>
</div>
`;
